org/oppia/proto/v1/structure/state.proto

```diff
@@ -0,0 +1,1170 @@
+  Outcome default_outcome = 2;
+
+  // Continue interactions cannot have custom answer groups, and do not support solutions. Users
+  // cannot receive hints since only one answer is possible (clicking the button).
+
+  // Represents the customization arguments available for instances of the continue interaction.
+  message CustomizationArgs {
+    option (org.oppia.proto.v1.versions.structure_proto_version_type) = STATE_PROTO_VERSION;
+
+    // Custom text to use for the 'Continue' button displayed for this interaction rather than using
+    // the button's default text ('Continue').
+    SubtitledText button_text = 1;
+  }
+}
+
+// Represents an interaction where the user is expected to input a textual fraction. Details:
+// - Answer type: Fraction
+// - Has a default outcome: yes
+// - Has support for showing hints: yes
+// - Has support for showing a solution: yes
+message FractionInputInstance {
+  option (org.oppia.proto.v1.versions.structure_proto_version_type) = STATE_PROTO_VERSION;
+
+  // The customization arguments for this interaction.
+  CustomizationArgs customization_args = 1;
+
+  // The answer groups for this interaction.
+  repeated AnswerGroup answer_groups = 2;
+
+  // The default outcome for this interaction.
+  Outcome default_outcome = 3;
+
+  // The hints corresponding to this interaction.
+  repeated Hint hints = 4;
+
+  // The solution corresponding to this interaction.
+  Solution solution = 5;
+
+  // Represents the customization arguments available for instances of the fraction input
+  // interaction.
+  message CustomizationArgs {
+    option (org.oppia.proto.v1.versions.structure_proto_version_type) = STATE_PROTO_VERSION;
+
+    // Specifics whether fractions must be provided in simplest form (that is, '2/4' would not be
+    // allowed, but '1/2' would be).
+    bool requires_simplest_form = 1;
+
+    // Specifics whether to allow improper fractions (i.e. whether fractions like '5/4' would be
+    // allowed).
+    bool allow_improper_fractions = 2;
+
+    // Specifics whether the fraction input can contain an integer part, that is, whether it can be
+    // a mixed fraction like '1 2/3'.
+    bool allow_nonzero_integer_part = 3;
+
+    // Specifies the placeholder/hint text that should be shown in the fraction input box prior to
+    // the user starting to enter an answer. Note that this will override the client's default text.
+    SubtitledText placeholder = 4;
+  }
+
+  // Represents the specific solution available to instances of the fraction input interaction.
+  message Solution {
+    option (org.oppia.proto.v1.versions.structure_proto_version_type) = STATE_PROTO_VERSION;
+
+    // The base properties of the solution.
+    BaseSolution base_solution = 1;
+
+    // The correct answer for this interaction that should be shown to the user.
+    Fraction correct_answer = 2;
+  }
+
+  // Represents the specific answer group used by instances of the fraction input interaction.
+  message AnswerGroup {
```

Comment:
Hi @BenHenning -- per our discussion today, maybe we might need to call this ApiAnswerGroup, ApiHint, etc. so that we can avoid having two classes in the same web python module which refer to "Hint" (one being the plain Python object class and the other being the proto representation).

Tbh though -- I don't like the name. The API is the thing that this data interacts with, and not a description of the data. It's also kind of generic (we have e.g. the client/server API on Web).

So -- what do you think about AnswerGroupDTO (for ["data transfer object"](https://en.wikipedia.org/wiki/Data_transfer_object)) instead? From the Wikipedia page: "The difference between data transfer objects and business objects or data access objects is that a DTO does not have any behavior except for storage, retrieval, serialization and deserialization of its own data (mutators, accessors, parsers and serializers). In other words, DTOs are simple objects that should not contain any business logic but may contain serialization and deserialization mechanisms for transferring data over the wire." -- would you say that that is a good description of what this is? (This contrasts with a "value object" that is more like our Python/JS domain objects.)

/cc @anandwana001

Reply:
DTO sounds good, but per camel casing rules it would be 'AnswerGroupDto'.
Does that look good as well?

Reply:
Yup!